src/plugins/process/process.c

```diff
@@ -0,0 +1,243 @@
+/**
+ * @file
+ *
+ * @brief Source for process plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "process.h"
+
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbinvoke.h>
+#include <kdbpluginprocess.h>
+
+#include <stdio.h>
+
+typedef struct
+{
+	ElektraPluginProcess * pp;
+	ElektraInvokeHandle * plugin;
+	Key * pluginName;
+	KeySet * pluginConfig;
+} Process;
+
+static void cleanup (Plugin * handle, Process * process, Key * errorKey)
+{
+	if (process->plugin) elektraInvokeClose (process->plugin, errorKey);
+	keyDel (process->pluginName);
+	ksDel (process->pluginConfig);
+	elektraFree (process);
+	elektraPluginSetData (handle, NULL);
+}
+
+int elektraInvoke1Arg (ElektraInvokeHandle * handle, const char * elektraPluginFunctionName, Key * k)
+{
+	if (!handle || !elektraPluginFunctionName) return -2;
+
+	// If we cast this right away although the function wasn't found it will cause a deadlock
+	const void * rawFunc = elektraInvokeGetFunction (handle, elektraPluginFunctionName);
+
+	if (!rawFunc) return -2;
```

Comment:
No error set?

Reply:
no it just crashed right away and when i separated the cast from getting the pointer it worked - so that was more fixed by trial and error ;)